core/app/models/spree/variant.rb

```diff
@@ -350,6 +350,15 @@ def variant_properties
       end.flatten.compact
     end
 
+    # An enumerable of media that goes along with a variant
+    # Stores may not necessarily want to attach images to variants
+    # This provides a more robust interface
+    #
+    # @return [Enumerable] of media for a variant
+    def gallery
+      Variant.reflect_on_association(:images) ? association(:images).reader : []
```

Comment:
We define this association, so we shouldn't test for it. If this is to prepare for a future paperclip removal PR, it should be done later.